Refactored menu

Schultzer opened the pull request:
Refactored menu.

- Removed Deselect from edit menu
- ~~Added support for `removeSelected` in Delete~~
- Moved window menu into `windowMenuTemplate`
- Moved help menu into `helpMenuTemplate`
- Moved app menu into `appMenuTemplate`

now we only append special menu items in `background.js` ~~it would be nice if we could move `quite` and `delete` menu items into there respectively templates for readability.~~

Schultzer committed:
Refactored menu

Refactored menu.

- Removed Deselect from edit menu
- Added support for `removeSelected` in Delete
- Moved window menu into `windowMenuTemplate`
- Moved help menu into `helpMenuTemplate`
- Moved app menu into `appMenuTemplate`

now we only append special menu items in `background.js` it would be nice if we could move `quite` and `delete` menu items into there respectively templates for readability.

danschultzer commented on src/app.js:
```diff
@@ -42,13 +42,14 @@ document.addEventListener('DOMContentLoaded', function () {
         deselectAll: function() { global.fileList.Select.deselectAll(); },
         selectUp: function(event) { global.fileList.Select.moveDirection('up', !event.shiftKey); },
         selectDown: function(event) { global.fileList.Select.moveDirection('down', !event.shiftKey); },
+        // removeSelected: function() { global.fileList.Select.removeSelected(); },
```
There shouldn't be comments like this.

danschultzer commented on src/app.js:
```diff
@@ -42,13 +42,14 @@ document.addEventListener('DOMContentLoaded', function () {
           }
         },
-        handleCmdOrCtrlBackspace: function(event) {
+        handleCmdOrCtrlDelete: function(event) {
```
Why change the name to delete when keycode 8 (backspace) is still used?

Schultzer replied:
On Mac backspace is called delete.
Backspace/delete keycode is 8.

danschultzer replied:
It's still called backspace: https://css-tricks.com/snippets/javascript/javascript-keycodes/

Delete is `46`.

danschultzer commented on src/app.js:
```diff
@@ -103,7 +104,8 @@ document.addEventListener('DOMContentLoaded', function () {
 
 document.addEventListener('copy', copySelectedToClipboard, true);
-document.addEventListener('deselectAll', function() { global.fileList.Select.deselectAll(); }, true);
+document.addEventListener('removeSelected', function() { global.fileList.Select.removeSelected(); }, true);
```
This could also be condensed to `document.addEventListener('removeSelected', global.fileList.Select.removeSelected, true);`

danschultzer reviewed (changes requested):
There's other things in this PR, I'll look at later.

Schultzer committed:
Removed line 45 and refactored line 106

danschultzer commented on src/app.js:
```diff
@@ -103,7 +103,8 @@ document.addEventListener('DOMContentLoaded', function () {
 
 document.addEventListener('copy', copySelectedToClipboard, true);
-document.addEventListener('deselectAll', function() { global.fileList.Select.deselectAll(); }, true);
+document.addEventListener('removeSelected', global.fileList.Select.removeSelected(), true);
```
This has to be `document.addEventListener('removeSelected', global.fileList.Select.removeSelected, true);`

danschultzer commented on src/background.js:
```diff
@@ -18,120 +21,37 @@ console.log('Settings thirdparty environment variables:', thirdparty_env);
 
 var mainWindow, editWindow, reportWindow;
 
-var setApplicationMenu = function () {
```
This whole setup is not good because you can't see how the whole menu is by just looking at the menu file. E.g. a menu file, the menu ends on a separator, and it makes no sense until you read it in this file, but it's not how it should be. Menus should be self contained. that's the reason the menu was here in the first place, and not in any specific file.

Schultzer committed:
Clean up

Moved help menu back into `background.js`
this is the only way I can make `shell.openExternal` work.

danschultzer commented on src/app.js:
```diff
@@ -103,7 +103,6 @@ document.addEventListener('DOMContentLoaded', function () {
 });
 
 document.addEventListener('copy', copySelectedToClipboard, true);
-document.addEventListener('deselectAll', function() { global.fileList.Select.deselectAll(); }, true);
```
Why not have this so you can deselect?

Schultzer replied:
We don't use that line.

danschultzer commented on src/background.js:
```diff
@@ -18,119 +20,23 @@ console.log('Settings thirdparty environment variables:', thirdparty_env);
         menus.push(devMenuTemplate);
     }
 
-    if (process.platform === 'darwin') {
```
Are you sure, we won't use other platforms?

Schultzer replied:
That's why we use [roles](https://github.com/electron/electron/blob/master/lib/browser/api/menu-item-roles.js).

danschultzer reviewed (changes requested):
Few things

Schultzer committed:
Named back to backspace

danschultzer committed:
Simplify menu templates

Schultzer committed:
Fix selectall

Schultzer committed:
Removed cmake
